internal/api/handlers/rooms_handler_test.go

```diff
@@ -0,0 +1,231 @@
+
+			rr := httptest.NewRecorder()
+			mux.ServeHTTP(rr, req)
+
+			require.Equal(t, 200, rr.Code)
+			bodyString := rr.Body.String()
+			require.Equal(t, "{\"success\":true}", bodyString)
+		}
+	})
+
+	t.Run("with valid request and nonexistent game room then it should return with status code 404", func(t *testing.T) {
+
+		mockCtrl := gomock.NewController(t)
+		defer mockCtrl.Finish()
+
+		clockMock := mock5.NewFakeClock(time.Now())
+		portAllocatorMock := mock2.NewMockPortAllocator(mockCtrl)
+		roomStorageMock := mock.NewMockRoomStorage(mockCtrl)
+		instanceStorageMock := mock3.NewMockGameRoomInstanceStorage(mockCtrl)
+		runtimeMock := mock4.NewMockRuntime(mockCtrl)
+
+		roomsManager := room_manager.NewRoomManager(clockMock, portAllocatorMock, roomStorageMock, instanceStorageMock, runtimeMock)
+
+		mux := runtime.NewServeMux()
+		err := api.RegisterRoomsServiceHandlerServer(context.Background(), mux, ProvideRoomsHandler(roomsManager))
+		require.NoError(t, err)
+
+		roomStorageMock.EXPECT().GetRoom(gomock.Any(), gomock.Any(), gomock.Any()).Return(nil, errors.NewErrNotFound("room not found in scheduler "))
+
+		request, err := validRawRequests[0].MarshalJSON()
+		require.NoError(t, err)
+
+		req, err := http.NewRequest(http.MethodPost, "/scheduler/scheduler-name-1/rooms/room-name-1/ping", bytes.NewReader(request))
+		require.NoError(t, err)
+
+		rr := httptest.NewRecorder()
+		mux.ServeHTTP(rr, req)
+
+		require.Equal(t, 404, rr.Code)
+	})
+
+	t.Run("with valid request when have error while updating game room then it should return with status code 500", func(t *testing.T) {
+
+		mockCtrl := gomock.NewController(t)
+		defer mockCtrl.Finish()
+
+		clockMock := mock5.NewFakeClock(time.Now())
+		portAllocatorMock := mock2.NewMockPortAllocator(mockCtrl)
+		roomStorageMock := mock.NewMockRoomStorage(mockCtrl)
+		instanceStorageMock := mock3.NewMockGameRoomInstanceStorage(mockCtrl)
+		runtimeMock := mock4.NewMockRuntime(mockCtrl)
+
+		roomsManager := room_manager.NewRoomManager(clockMock, portAllocatorMock, roomStorageMock, instanceStorageMock, runtimeMock)
+
+		mux := runtime.NewServeMux()
+		err := api.RegisterRoomsServiceHandlerServer(context.Background(), mux, ProvideRoomsHandler(roomsManager))
+		require.NoError(t, err)
+
+		roomStorageMock.EXPECT().GetRoom(gomock.Any(), gomock.Any(), gomock.Any()).Return(gameRoom, nil)
+		roomStorageMock.EXPECT().UpdateRoom(gomock.Any(), gomock.Any()).Return(errors.NewErrUnexpected("error updating room on redis"))
+
+		request, err := validRawRequests[0].MarshalJSON()
+		require.NoError(t, err)
+
+		req, err := http.NewRequest(http.MethodPost, "/scheduler/scheduler-name-1/rooms/room-name-1/ping", bytes.NewReader(request))
+		require.NoError(t, err)
+
+		rr := httptest.NewRecorder()
+		mux.ServeHTTP(rr, req)
+
+		require.Equal(t, 500, rr.Code)
+	})
+
+	t.Run("with valid request when the new game room state transition is invalid then it should return with status code 500", func(t *testing.T) {
```

Comment:
InternalServerError or Conflict? Maybe a UnprocessableEntity... is this taking into count the current maestro ping contract?

Reply:
maestro-client currently does not consider the response status code, it would make sense to return a UnprocessableEntity code in this case, but I wasn't able to find an equivalent status code on the GRPC protocol which we are using.

Reply:
mestro-client code that i looked upon:
![image](https://user-images.githubusercontent.com/33766735/131550499-7792c38f-4d70-4aa3-9163-6b8489e959eb.png)